Source/msg.cpp

```diff
@@ -275,13 +275,13 @@ void DeltaImportJunk(byte *src)
 	}
 
 	int q = 0;
-	for (int i = 0; i < MAXQUESTS; i++) {
-		if (!QuestData[i].isSinglePlayerOnly) {
+	for (auto &quest : Quests) {
+		if (!QuestData[q].isSinglePlayerOnly) {

```

Comment:
Since `q` is only incremented when `QuestData[i].isSinglePlayerOnly` is `false` we can't use it as the index as it will stop to match `i` the first time `isSinglePlayerOnly` is true.

Instead you could use `quest._idx`

Reply:
Updated to use `quest._idx`. Thank you!

Reply:
Note this is present in several places in this pr

Reply:
Updated the pull request to use `quest._qidx` except in `void InitQuests()` where we have not initialized `quest._qidx` yet. Thank you!